src/components/AddTask/AddTask.tsx

```diff
@@ -0,0 +1,82 @@
+import "./AddTask.css";
+import addButton from "../../assets/add-button.svg";
+import TaskListContainer from "../Task/TaskList/TaskListContainer";
+import { useForm, SubmitHandler } from "react-hook-form";
+import { yupResolver } from "@hookform/resolvers/yup";
+import * as yup from "yup";
+
+interface Task {
+  id: number;
+  name: string;
+  completed: boolean;
+}
+
+interface AddTaskProps {
+  tasks: Task[];
+  addTask: (task: Task) => void;
+}
+
+const schema = yup
+  .object({
+    taskName: yup
+      .string()
+      .required("Task name is required")
+      .min(3, "Task name must be at least 3 characters long"),
+  })
+  .required();
+
+const AddTask = ({ tasks, addTask }: AddTaskProps): JSX.Element => {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    reset,
+  } = useForm<{ taskName: string }>({
+    resolver: yupResolver(schema),
+  });
+
+  // const [taskName, setTaskName] = useState<string>("");
+
+  const onSubmit: SubmitHandler<{ taskName: string }> = (data) => {
```

Comment:
Why we need to use SubmitHandler? can we use hook form handleSubmit?

Reply:
I am using handleSubmit. since I am using typescript , Submithandler is for type safety in typescript and to give the form that data type